### PR TITLE
Created main router

### PR DESCRIPTION
**Pull request by Mellifaro**

### CODE REVIEW

File: services/general-routing/generalrouting-impl/pom.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>general-routing</artifactId>
```
Comment (Dominys) [inline services/general-routing/generalrouting-impl/pom.xml]: @Mellifaro WTF is general-routing and for what is this ?

Comment (Mellifaro) [reply services/general-routing/generalrouting-impl/pom.xml]: Maybe, I misunderstood the architecture of application. We have one main router(module "general-routing"). It parses all our request by type. If type is "authorization" we delegate our request to AuthorizationRouter, that then delegates this request to one of handlers. If type is "profile" we delegate our request to ProfileRouter. I don't have other idea, how to make routing by type without main router. Now I see such hierarchy: (main_router) -> (AuthRouter, ProfileRouter, MessengerRouter) - > (A lot of handlers...). I also see the variant to make main routing later in rest contollers. What variant is right?

Comment (Dominys) [reply services/general-routing/generalrouting-impl/pom.xml]: @Mellifaro Leave it in router module. Or make router-impl module.

**Comment by Mellifaro** [comment]
Moved main router to module "common-router".

File: basis/common-router/src/main/java/com/softgroup/common/router/api/mainRouterImpl/MainRouterHandlerFactory.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.softgroup.common.router.api.factories.AbstractHandlerFactory;
+import com.softgroup.common.router.api.routers.TypeRouterHandler;
+import org.springframework.stereotype.Component;
+
+/**
+ * Created by Виктор on 13.03.2017.
+ */
+@Component
+public class MainRouterHandlerFactory extends AbstractHandlerFactory<TypeRouterHandler> {
```
Comment (Dominys) [inline basis/common-router/src/main/java/com/softgroup/common/router/api/mainRouterImpl/MainRouterHandlerFactory.java]: @Mellifaro this must be in impl. Not in api.

File: basis/common-router/src/main/java/com/softgroup/common/router/api/mainRouterImpl/MainRouterHandlerImpl.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import com.softgroup.common.router.api.AbstractRouterHandler;
+import com.softgroup.common.router.api.routers.MainRouterHandler;
+import com.softgroup.common.router.api.routers.TypeRouterHandler;
+import org.springframework.stereotype.Component;
+
+/**
+ * Created by Виктор on 13.03.2017.
+ */
+@Component
+public class MainRouterHandlerImpl extends AbstractRouterHandler<TypeRouterHandler>
```
Comment (Dominys) [inline basis/common-router/src/main/java/com/softgroup/common/router/api/mainRouterImpl/MainRouterHandlerImpl.java]: @Mellifaro Why this is in api ??

File: services/pom.xml [hunk 4]
```diff
@@ -14,6 +14,7 @@
 	<modules>
 
 		<module>authorization</module>
+		<module>general-routing</module>
```
Comment (Dominys) [inline services/pom.xml]: @Mellifaro WTF is general-routing ??

File: basis/common-router/src/main/java/com/softgroup/common/router/api/mainRouterImpl/MainRouterHandlerFactory.java (hunk 2 above)
Comment (Mellifaro) [reply basis/common-router/src/main/java/com/softgroup/common/router/api/mainRouterImpl/MainRouterHandlerFactory.java]: ok. I will correct it

**Comment by Mellifaro** [comment]
I decided, that it will be better to move main-router module to the services module.

**Comment by Dominys** [comment]
@Mellifaro Router is not service

**Review by Dominys — APPROVED** [review]

**Pull request merged**